libsawtooth/src/publisher/factory.rs

```diff
@@ -0,0 +1,160 @@
+ * limitations under the License.
+ * ------------------------------------------------------------------------------
+ */
+
+//! Conatains the struct for starting publishing of a new artifact
+
+use std::sync::mpsc::channel;
+use std::thread;
+
+use crate::artifact::{Artifact, ArtifactCreator, ArtifactCreatorFactory};
+use crate::error::InternalError;
+
+use super::message::PublishMessage;
+use super::{
+    Batch, BatchExecutionResult, BatchVerifierFactory, PendingBatches, PublishHandle,
+    PublishingContext,
+};
+
+/// Factory to start publishing a new artificat
+///
+/// The `PublishFactory` starts up the exeuction and verification of batches by
+/// spinning up a thread and returning a `PublishHandle` that can be used to
+/// control the thread execution.
+///
+// allow type complexity here because of the use of associated types
+#[allow(clippy::type_complexity)]
+pub struct PublishFactory<B, C, R, I>
+where
+    B: 'static + Batch + Clone,
+    C: 'static + PublishingContext + Clone,
+    R: 'static + Artifact,
+    I: 'static + BatchExecutionResult,
+{
+    artifact_creator_factory: Box<
+        dyn ArtifactCreatorFactory<
+            ArtifactCreator = Box<dyn ArtifactCreator<Context = C, Input = Vec<I>, Artifact = R>>,
+        >,
+    >,
+    batch_verifier_factory:
+        Box<dyn BatchVerifierFactory<Batch = B, Context = C, ExecutionResult = I>>,
+}
+
+/// Create a new `PublishFactory`
+///
+/// Arguments
+///
+/// * `artifact_creator_factory` - a factory for creating a new artifact creator for a new
+///    publishing thread
+/// * `batch_verifier_factory` - a factory for creating a new batch verifier for a new publishing
+///    thread
+///
+// allow type complexity here because of the use of associated types
+#[allow(clippy::type_complexity)]
+impl<B, C, R, I> PublishFactory<B, C, R, I>
+where
+    B: 'static + Batch + Clone,
+    C: 'static + PublishingContext + Clone,
+    R: 'static + Artifact,
+    I: 'static + BatchExecutionResult,
+{
+    pub fn new(
+        artifact_creator_factory: Box<
+            dyn ArtifactCreatorFactory<
+                ArtifactCreator = Box<
+                    dyn ArtifactCreator<Context = C, Input = Vec<I>, Artifact = R>,
+                >,
+            >,
+        >,
+        batch_verifier_factory: Box<
+            dyn BatchVerifierFactory<Batch = B, Context = C, ExecutionResult = I>,
+        >,
+    ) -> Self {
+        Self {
+            artifact_creator_factory,
+            batch_verifier_factory,
+        }
+    }
+}
+
+impl<B, C, R, I> PublishFactory<B, C, R, I>
+where
+    B: 'static + Batch + Clone,
+    C: 'static + PublishingContext + Clone,
+    R: 'static + Artifact,
+    I: 'static + BatchExecutionResult,
+{
+    /// Start building the next publishable unit, referred to as a block going forward
+    /// The publisher will start pulling batches off of a pending queue for the provided service
+    /// and
```

Comment:
Unfinished comment

Reply:
The whole comment was actually outdated as well. Fixed.